### PR TITLE
Add website.i2pd.i2pd

### PR DESCRIPTION
**Pull request by l-n-s**

### CODE REVIEW

File: website.i2pd.i2pd.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2018 Viktor Villainov <supervillain@riseup.net> -->
+<component type="desktop">
+  <id>i2pd.desktop</id>
```
Comment (nedrichards) [inline website.i2pd.i2pd.appdata.xml]: this needs to be the app-id.

**Comment by nedrichards** [comment]
bot, build website.i2pd.i2pd

File: website.i2pd.i2pd.appdata.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2018 Viktor Villainov <supervillain@riseup.net> -->
 <component type="desktop">
-  <id>i2pd.desktop</id>
+  <id>website.i2pd.i2pd.desktop</id>
```
Comment (bilelmoussaoui) [inline website.i2pd.i2pd.appdata.xml]: No need for `.desktop` here.

Comment (l-n-s) [reply website.i2pd.i2pd.appdata.xml]: ok, just took signal manifest as example. will fix it

**Comment by l-n-s** [comment]
fixed the url to miniupnpc source, iy shouldn't fail this time

**Comment by nedrichards** [comment]
bot, build website.i2pd.i2pd

File: website.i2pd.i2pd.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2018 Viktor Villainov <supervillain@riseup.net> -->
+<component type="desktop">
+  <id>website.i2pd.i2pd</id>
```
Comment (TingPing) [inline website.i2pd.i2pd.appdata.xml]: Below this add `<launchable type="desktop-id">website.i2pd.i2pd.desktop</launchable>`

File: website.i2pd.i2pd.json [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.11",
+    "sdk": "org.kde.Sdk",
+    "command": "i2pd_qt_wrapper",
+    "finish-args": [
+        "--socket=x11",
+        "--socket=wayland",
+        "--share=ipc",
+        "--share=network",
+        "--device=dri"],
```
Comment (TingPing) [inline website.i2pd.i2pd.json]: `]` on newline.

File: website.i2pd.i2pd.json [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                    "sha256": "7f6130bc3cf65f56a618888ce9d5ea704fa10b462be126ad053e80e553d6d8b7"
+                }
+            ]
+        },
+        {
+            "name": "openssl",
+            "buildsystem": "simple",
+            "build-commands": [
+                "./config --prefix=/app --openssldir=/app/ssl shared zlib",
+                "make",
```
Comment (TingPing) [inline website.i2pd.i2pd.json]:
```suggestion
                "make -j $FLATPAK_BUILDER_N_JOBS",
```

File: website.i2pd.i2pd.json [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://dl.bintray.com/boostorg/release/1.68.0/source/boost_1_68_0.tar.bz2",
+                    "sha256": "7f6130bc3cf65f56a618888ce9d5ea704fa10b462be126ad053e80e553d6d8b7"
+                }
+            ]
+        },
+        {
+            "name": "openssl",
```
Comment (TingPing) [inline website.i2pd.i2pd.json]: Could you perhaps put a comment here pointing out that once the KDE runtime is based on FDO 18.08 (maybe KDE 5.12? dunno). This should be removed.

File: website.i2pd.i2pd.json [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+                {
+                    "type": "archive",
+                    "url": "https://www.openssl.org/source/openssl-1.1.1.tar.gz",
+                    "sha256": "2836875a0f89c03d0fdf483941512613a50cfb421d6fd94b9f41d7279d586a3d"
+                }
+            ]
+        },
+        {
+            "name": "miniupnpc",
+            "buildsystem": "simple",
```
Comment (TingPing) [inline website.i2pd.i2pd.json]: Don't use the `simple` build system here:

```json
{
  "name": miniupnpc",
  "no-autogen": true,
  "make-install-args": ["INSTALLPREFIX=/app"],
}
```

File: website.i2pd.i2pd.json (hunk 6 above)
Comment (l-n-s) [reply website.i2pd.i2pd.json]: It currently uses openssl v1.1.1 for the best performance. Sure, when upstream SDK will support 1.1.1, it will be removed.

**Comment by nedrichards** [comment]
bot, build website.i2pd.i2pd

**Comment by bilelmoussaoui** [comment]
The appdata file, desktop file are in upstream too https://github.com/PurpleI2P/i2pd/commit/929a27a5acd6403eac5a6e52a0a32c7c35e0edb3 you could update them there and push a new version. Same goes for icons. It would be great if they are installed using the current build system. You could use some other sizes for the icons too to make sure the app icon is looking good on every screen size/software center... :)

**Comment by l-n-s** [comment]
Can somebody explain what that means? https://flathub.org/builds/#/builders/30/builds/701/steps/6/logs/stdio

File: website.i2pd.i2pd.json [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+                }
+            ],
+            "post-install": [
+                "install -Dm777 i2pd_qt /app/bin/i2pd_qt",
+                "install -Dm777 ../../i2pd_qt_wrapper /app/bin/i2pd_qt_wrapper",
+                "install -Dm644 ../../contrib/i2pd.conf /app/i2pd/i2pd.conf",
+                "install -Dm644 ../../contrib/tunnels.conf /app/i2pd/tunnels.conf",
+                "cp -r ../../contrib/certificates /app/i2pd/certificates",
+                "install -Dm644 resources/images/icon.png /app/share/icons/hicolor/128x128/apps/website.i2pd.i2pd.png",
+                "install -Dm644 ../../website.i2pd.i2pd.appdata.xml /app/share/applications/website.i2pd.i2pd.appdata.xml",
```
Comment (TingPing) [inline website.i2pd.i2pd.json]:
```suggestion
                "install -Dm644 ../../website.i2pd.i2pd.appdata.xml /app/share/metainfo/website.i2pd.i2pd.appdata.xml",
```

@l-n-s 	This is why the build failed

Comment (l-n-s) [reply website.i2pd.i2pd.json]: arrite now it's fixed

**Comment by nedrichards** [comment]
bot, build website.i2pd.i2pd

**Comment by nedrichards** [comment]
Build succesful: https://flathub.org/builds/#/builders/2/builds/5778

**Comment by l-n-s** [comment]
Good. I've updated the upstream appdata and desktop files + added resized icons. The next release will be next month or so.

**Comment by nedrichards** [comment]
@l-n-s is there anyone else from upstream who needs access to the repo?

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/website.i2pd.i2pd

**Comment by l-n-s** [comment]
Great! @nedrichards yes, please invite @r4sas

**Comment by r4sas** [comment]
@nedrichards can you resend invation? I dont have permissions on that repo :eyes:

**Comment by barthalion** [comment]
@r4sas Please check here: https://github.com/flathub/website.i2pd.i2pd/invitations

**Comment by r4sas** [comment]
@barthalion thank you, now received it.

**Pull request closed without merging**